compiler/packages/babel-plugin-react-compiler/src/Inference/InferEffectDependencies.ts

```diff
@@ -196,24 +202,28 @@ export function inferEffectDependencies(fn: HIRFunction): void {
              */
 
             const usedDeps = [];
-            for (const dep of minimalDeps) {
+            for (const maybeDep of minimalDeps) {
               if (
-                ((isUseRefType(dep.identifier) ||
-                  isSetStateType(dep.identifier)) &&
-                  !reactiveIds.has(dep.identifier.id)) ||
-                isFireFunctionType(dep.identifier)
+                ((isUseRefType(maybeDep.identifier) ||
+                  isSetStateType(maybeDep.identifier)) &&
+                  !reactiveIds.has(maybeDep.identifier.id)) ||
+                isFireFunctionType(maybeDep.identifier)
               ) {
                 // exclude non-reactive hook results, which will never be in a memo block
                 continue;
               }
 
-              const {place, instructions} = writeDependencyToInstructions(
+              const dep = truncateDepAtCurrent(maybeDep);
```

Comment:
Moved this check from `writeDependencyToInstructions` so that IDE highlighting and etc also receive the correct (truncated) dependency

Reply:
Nice, thank you!

IIUC, the IDE has access to all of the paths but we'll still need to look inside the effect body for property accesses/optional chains that correspond to deps, is that right?

And #32099 would make this easier because it would hoist the deps into temporaries that would be re-used inside the body.